app/jobs/notifications/moderation_notification_job.rb

```diff
@@ -0,0 +1,22 @@
+module Notifications
+  class ModerationNotificationJob < ApplicationJob
+    queue_as :send_moderation_notification
+
+    def perform(notifiable_id, service = Notifications::Moderation::Send)
+      available_moderators = Notifications.available_moderators
+      return if available_moderators.empty?
+
+      moderator = self.class.random_moderator(available_moderators)
+
+      # notifiable is currently only comment
+      notifiable = Comment.find_by(id: notifiable_id)
+      return unless notifiable
+
+      service.call(moderator, notifiable)
+    end
+
+    def self.random_moderator(available_moderators)
+      available_moderators.order(Arel.sql("RANDOM()")).first
```

Comment:
why not just inline this at line 9?

BTW if you want to reduce the two SQL calls (check if exists and then get the random one) to one, you can just ask the DB to give you a random available moderator and then check if it's not nil.